Subject: details: Fix showing the firmware type in case of firmware auto selection
From: Lin Ma lma@suse.com Wed Jan 15 10:34:12 2020 +0800
Date: Wed Jan 15 10:21:46 2020 -0500:
Git: 15a9502b7b7a263c4d66ff2b3f31c209f58fe0b4

For a shutoff VM, If user select uefi firmware auto selection, e.g.
<os firmware='efi'>
...
</os>

Its firmware information is set to 'BIOS' in details, This is incorrect.
This fixes it.

Reviewed-by: Cole Robinson <crobinso@redhat.com>
Signed-off-by: Lin Ma <lma@suse.com>

Index: virt-manager-2.2.1/virtManager/details/details.py
===================================================================
--- virt-manager-2.2.1.orig/virtManager/details/details.py
+++ virt-manager-2.2.1/virtManager/details/details.py
@@ -2017,8 +2017,11 @@ class vmmDetails(vmmGObjectUI):
 
         # Firmware
         domcaps = self.vm.get_domain_capabilities()
-        firmware = domcaps.label_for_firmware_path(
-            self.vm.get_xmlobj().os.loader)
+        if self.vm.get_xmlobj().is_uefi():
+            firmware = 'UEFI'
+        else:
+            firmware = domcaps.label_for_firmware_path(
+                self.vm.get_xmlobj().os.loader)
         if self.widget("overview-firmware").is_visible():
             uiutil.set_list_selection(
                 self.widget("overview-firmware"), firmware)
